src/DataTypes/IDataType.cpp

```diff
@@ -318,14 +322,15 @@ SerializationPtr IDataType::getDefaultSerialization(SerializationPtr override_de
     return doGetDefaultSerialization();
 }
 
-SerializationPtr IDataType::getSerialization(ISerialization::KindStack kind_stack, SerializationPtr override_default) const
+SerializationPtr IDataType::getSerialization(
+    ISerialization::KindStack kind_stack, const SerializationInfoSettings & settings, SerializationPtr override_default) const
 {
     auto serialization = getDefaultSerialization(override_default);
     for (auto kind : kind_stack)
     {
-        if (supportsSparseSerialization() && kind == ISerialization::Kind::SPARSE)
+        if (settings.canUseSparseSerialization(*this) && kind == ISerialization::Kind::SPARSE)
             serialization = std::make_shared<SerializationSparse>(serialization);
-        else if (kind == ISerialization::Kind::DETACHED)
```

Comment:
Any reason for removing the `else` except that if you take the previous if you won't take the second one?